src/Tests/Project/JobRunner.test.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assert} from 'chai';
+
+import {JobRunner} from '../../Project/JobRunner';
+import {WorkJobs} from '../../Project/WorkJobs';
+import {obtainWorkspaceRoot} from '../../Utils/Helpers';
+import {Logger} from '../../Utils/Logger';
+import {MockJob} from '../MockJob';
+
+suite('Project', function() {
+  suite('JobRunner', function() {
+    const logger = new Logger();
+    suite('#start()', function() {
+      test('jobs are done', function(done) {
+        let jobRunner = new JobRunner(logger);
+        const workspaceRoot: string = obtainWorkspaceRoot();
+        let workJobs = new WorkJobs();
+        workJobs.push(new MockJob('mockup'));
+        workJobs.push(new MockJob('mockup'));
+        assert.strictEqual(workJobs.length, 2);
+
+        // overwrite events so that multiple events will be emitted
```

Comment:
It seems this test does several checks within one test,
(1) adding two mockup jobs and cehck if workJobs has length 2
(2) cleaning up the workJob queue and check if workJobs has length 0 as a result

Do I understand alright? If I do, what about separating them into two tests?

Reply:
1. This test is for `JobRunner`.
2. `adding two mockup jobs and cehck` is a sub-process of JobRunner so that I added
3. If new test for adding jobs is introduced, it would be in `WorkJobs`.
4. But, WorkJobs is a Array<Job> so that we don't need to test it.

Reply:
@YongseopKim 

> 2. adding two mockup jobs and cehck is a sub-process of JobRunner so that I added

It sounds to be optional. It seems it aims to test mockUp addition and checking 'cleaning' working. If it's not about JobRunner's test, separating them and renaming as mockup would seems to make this test much clearer, IMO.

Reply:
Is it resolved?

Reply:
I looked up the code and it still looks like a testing code for a 'EventEmitter' rather than the JobRunner class itself. I find this test still vague in purpose. I'd rather recommend another test, but as you may. I am not strongly against it.

Reply:
If so, let's have time to talk about this test soon. 😁